lib/node_modules/@stdlib/math/base/special/gammaln/test/test.js

```diff
@@ -87,31 +87,49 @@ tape( 'the function returns `-ln(x)` for very small x', function test( t ) {
 });
 
 tape( 'the function evaluates the natural logarithm of the gamma function (positive integers)', function test( t ) {
+	var expected;
 	var delta;
 	var tol;
-	var v;
+	var x;
+	var y;
 	var i;
 
-	for ( i = 0; i < data1.length; i++ ) {
-		v = gammaln( data1[ i ] );
-		delta = abs( v - expected1[ i ] );
-		tol = 4.75e-15 * max( 1.0, abs( v ), abs( expected1[ i ] ) );
-		t.ok( delta <= tol, 'within tolerance. x: ' + data1[ i ] + '. Value: ' + v + '. Expected: ' + expected1[ i ] + '. Tolerance: ' + tol + '.' );
+	x = data1;
+	expected = expected1;
+
+	for ( i = 0; i < x.length; i++ ) {
+		y = gammaln( x[ i ] );
+		if ( y === expected[ i ] ) {
+			t.equal( y, expected[ i ], 'x: '+x[i]+'. Expected: '+expected[i] );
+		} else {
+			delta = abs( y - expected[ i ] );
+			tol = 21.2 * EPS * abs( expected[ i ] );
+			t.ok( delta <= tol, 'within tolerance. x: '+x[i]+'. y: '+y+'. E: '+expected[i]+'. tol: '+tol+'. Δ: '+delta+'.' );
+		}
 	}
 	t.end();
 });
 
 tape( 'the function evaluates the natural logarithm of the gamma function (decimal values)', function test( t ) {
+	var expected;
 	var delta;
 	var tol;
-	var v;
+	var x;
+	var y;
 	var i;
 
-	for ( i = 0; i < data2.length; i++ ) {
-		v = gammaln( data2[ i ] );
-		delta = abs( v - expected2[ i ] );
-		tol = 2.5e-12 * max( 1.0, abs( v ), abs( expected2[ i ] ) );
-		t.ok( delta <= tol, 'within tolerance. x: ' + data2[ i ] + '. Value: ' + v + '. Expected: ' + expected2[ i ] + '. Tolerance: ' + tol + '.' );
+	x = data2;
+	expected = expected2;
+
+	for ( i = 0; i < x.length; i++ ) {
+		y = gammaln( x[ i ] );
+		if ( y === expected[ i ] ) {
+			t.equal( y, expected[ i ], 'x: '+x[i]+'. Expected: '+expected[i] );
+		} else {
+			delta = abs( y - expected[ i ] );
+			tol = 1.1e4 * EPS * abs( expected[ i ] );
```

Comment:
@Planeshifter This is also a rather large tolerance. I wonder if the issue is R. Would be good to compare against Julia and NumPy/SciPy. Or more perhaps even more ideally, we'd compare against an arbitrary precision implementation.

Reply:
For `gamma`, we use Cephes. In this case, looks like we use `fdlibm`.

Reply:
Investigated and found that Julia is way closer to our implementation (`8.5 * EPS` to get tests to pass.)